lib/rides/models/ride.dart

```diff
@@ -126,6 +126,12 @@ class Ride extends Trip {
     return false;
   }
 
+  @override
+  bool shouldShowInListView({required bool past}) {
+    return super.shouldShowInListView(past: past) &&
+        (!past || (status != RideStatus.pending && status != RideStatus.withdrawnByRider));
```

Comment:
RideStatus.withdrawnByRider implies hideInListView. Since it should change both when done. If we say we still leave it in for safty then I would recommend to also check for RideStatus.Preview

Reply:
Yes I know, I just wanted to make this explicit. I could add preview here, but it would be confusing if you consider the Search-"ListView" where it is shown.